### PR TITLE
Json Validation and Mapping

### PR DESCRIPTION
**Pull request by ChandraAddala**
- Add java bean validation to some /events and /search endpoints using hibernate validator
 - Made all error responses in consistent format
 - Added validation for /events when we get multiple root elements

### CODE REVIEW

File: blueflood-core/src/main/java/com/rackspacecloud/blueflood/inputs/formats/AggregatedPayload.java [hunk 1]
```diff
@@ -45,7 +45,7 @@
 
     @EpochRange(maxPast = EpochRangeLimits.BEFORE_CURRENT_TIME_MS,
             maxFuture = EpochRangeLimits.AFTER_CURRENT_TIME_MS,
-            message = "Out of bounds. Cannot be more than ${maxPast.getValue()} milliseconds into the past. Cannot be more than ${maxPast.getValue()} milliseconds into the future")
+            message = "Out of bounds. Cannot be more than ${maxPast.getValue()} milliseconds into the past. Cannot be more than ${maxFuture.getValue()} milliseconds into the future")
```
Comment (shintasmith) [inline blueflood-core/src/main/java/com/rackspacecloud/blueflood/inputs/formats/AggregatedPayload.java]: 👍 good fix

File: blueflood-http/src/test/java/com/rackspacecloud/blueflood/inputs/handlers/HttpEventsIngestionHandlerTest.java [hunk 2]
```diff
@@ -33,7 +38,7 @@
 import static org.mockito.Mockito.*;
 import static junit.framework.Assert.*;
 
-public class HttpEventsIngestionHandlerTest {
+public class HttpEventsIngestionHandlerTest extends BaseHandlerTest {
```
Comment (shintasmith) [inline blueflood-http/src/test/java/com/rackspacecloud/blueflood/inputs/handlers/HttpEventsIngestionHandlerTest.java]: I just noticed this... If you name a class *Test, the junit test runner will try to run it. If this is meant to be a base class, and not a test class, can we name it HandlerTestsBase ?

File: blueflood-http/src/main/java/com/rackspacecloud/blueflood/inputs/handlers/HttpEventsIngestionHandler.java [hunk 3]
```diff
@@ -65,40 +68,48 @@ public void handle(ChannelHandlerContext ctx, FullHttpRequest request) {
...
-                throw new InvalidDataException( HttpMetricsIngestionHandler.ERROR_HEADER + System.lineSeparator() + "'" + event.getWhat() + "': 'when' '" + timestamp + "' is more than '" + pastDiff + "' milliseconds into the past." );
+            if (!validationErrors.isEmpty()) {
+                DefaultHandler.sendErrorResponse(ctx, request, validationErrors, HttpResponseStatus.BAD_REQUEST);
+                return;
             }
 
             searchIO.insert(tenantId, Arrays.asList(event.toMap()));
+            DefaultHandler.sendResponse(ctx, request, response, HttpResponseStatus.OK);
         } catch (JsonMappingException e) {
             log.error(String.format("Exception %s", e.toString()));
```
Comment (usnavi) [inline blueflood-http/src/main/java/com/rackspacecloud/blueflood/inputs/handlers/HttpEventsIngestionHandler.java]: do we want to log BAD_REQUESTs as errors?  Do we need to log these at all?  We'll see the status code in our access logs.

File: blueflood-http/src/main/java/com/rackspacecloud/blueflood/outputs/handlers/HttpRollupsQueryHandler.java [hunk 4]
```diff
@@ -141,13 +148,13 @@ public void handle(ChannelHandlerContext ctx, FullHttpRequest request) {
         } catch (InvalidRequestException e) {
             // let's not log the full exception, just the message.
             log.warn(e.getMessage());
```
Comment (usnavi) [inline blueflood-http/src/main/java/com/rackspacecloud/blueflood/outputs/handlers/HttpRollupsQueryHandler.java]: Here we are logging BAD_REQUEST as warnings.  Same question about if we want to log this, and at what level.

**Comment by ChandraAddala** [comment]
@shintasmith @usnavi I had addressed your comments. Pls take a look

**Pull request merged**